### PR TITLE
hwmv2: microchip_mec: port to HWMv2

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Port the MEC soc and boards to hwmv2.

### CODE REVIEW

File: soc/microchip/mec/Kconfig [hunk 1]
```diff
@@ -216,7 +210,17 @@ config MCHP_HEADER_VERBOSE_OUTPUT
 
 endif # MCHP_MEC_UNSIGNED_HEADER
 
-# Select SoC Part No. and configuration options
-source "soc/soc_legacy/arm/microchip_mec/*/Kconfig.soc"
+config SOC_MEC172X_PROC_CLK_DIV
```
Comment (fabiobaltieri) [inline soc/microchip/mec/Kconfig:213]: This one is used by mec15xx drivers as well as it turns out, should probably be renamed but I just moved it up the hierarchy in the interest of not changing the matching usage in drivers/

File: soc/microchip/mec/Kconfig.soc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Microchip MEC172x, MEC1501 MCU line
+
+# Copyright (c) 2018 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_MEC
```
Comment (nordicjm) [inline soc/microchip/mec/Kconfig.soc:6]:
```suggestion
config SOC_FAMILY_MICROCHIP_MEC
```

File: soc/microchip/mec/Kconfig.soc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+	bool
+
+config SOC_FAMILY
+	string
```
Comment (nordicjm) [inline soc/microchip/mec/Kconfig.soc:10]:
```suggestion
```

File: soc/microchip/mec/Kconfig.soc [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	default "microchip_mec" if SOC_FAMILY_MEC
```
Comment (nordicjm) [inline soc/microchip/mec/Kconfig.soc:11]: Would need updating here and anywhere else, etc.

File: soc/microchip/mec/mec1501x/Kconfig [hunk 5]
```diff
@@ -3,15 +3,18 @@
...
 
-choice
-	prompt "MEC1501 Selection"
-	depends on SOC_SERIES_MEC1501X
+config SOC_SERIES_MEC1501X
+	select ARM
+	select CPU_CORTEX_M4
+	select CPU_CORTEX_M_HAS_DWT
+	select CPU_HAS_ARM_MPU
+	select SOC_FAMILY_MEC
```
Comment (nordicjm) [inline soc/microchip/mec/mec1501x/Kconfig:11]: `SOC_FAMILY_*` selection goes in `Kconfig.soc`

File: soc/microchip/mec/mec172x/Kconfig [hunk 6]
```diff
@@ -1,32 +1,23 @@
...
-choice
-	prompt "MEC172x Selection"
-	depends on SOC_SERIES_MEC172X
+config SOC_SERIES_MEC172X
+	select ARM
+	select CPU_CORTEX_M4
+	select CPU_CORTEX_M_HAS_DWT
+	select CPU_HAS_FPU
+	select CPU_HAS_ARM_MPU
+	select SOC_FAMILY_MEC
```
Comment (nordicjm) [inline soc/microchip/mec/mec172x/Kconfig:12]: as above

File: soc/microchip/mec/Kconfig (hunk 1 above)
Comment (nordicjm) [reply soc/microchip/mec/Kconfig:213]: Falls on the maintainers responsibility to fix

Comment (fabiobaltieri) [reply soc/microchip/mec/Kconfig:213]: yeah fine by me... I can also rename it here if you prefer, whatever, no difference but this change is needed for the boards to build

File: soc/microchip/mec/Kconfig.soc (hunk 2 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/Kconfig.soc]: done

File: soc/microchip/mec/Kconfig.soc (hunk 4 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/Kconfig.soc]: yup, renamed everywhere throughout the code base

File: soc/microchip/mec/mec1501x/Kconfig (hunk 5 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/mec1501x/Kconfig]: cool, it was there already, guess I forgot to delete it from here

File: soc/microchip/mec/mec172x/Kconfig (hunk 6 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/mec172x/Kconfig]: same, dropped

**Comment by fabiobaltieri** [comment]
Rebased as well, merge conflict on the doc index file.

File: soc/microchip/mec/Kconfig.soc (hunk 3 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/Kconfig.soc]: dropped

File: soc/microchip/mec/Kconfig [hunk 7]
```diff
@@ -216,7 +210,17 @@ config MCHP_HEADER_VERBOSE_OUTPUT
...
+	  This divisor defines a ratio between processor clock (HCLK)
+	  and main 96 MHz clock (MCK):
+	  HCLK = MCK / PROC_CLK_DIV
+	  Allowed divider values: 1, 3, 4, 16, and 48.
+
+endif # SOC_FAMILY_MICROCHIP_MEC
 
-endif # SOC_FAMILY_MEC
+# Select SoC Part No. and configuration options
+rsource "*/Kconfig"
```
Comment (nordicjm) [inline soc/microchip/mec/Kconfig:226]:
```suggestion
# Select SoC Part No. and configuration options
rsource "*/Kconfig"

endif # SOC_FAMILY_MICROCHIP_MEC
```

File: soc/microchip/mec/Kconfig.defconfig [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2024, Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+rsource "*/Kconfig.defconfig.series"
```
Comment (nordicjm) [inline soc/microchip/mec/Kconfig.defconfig:4]:
```suggestion
if SOC_FAMILY_MICROCHIP_MEC

rsource "*/Kconfig.defconfig.series"

endif # SOC_FAMILY_MICROCHIP_MEC
```

**Review by nordicjm — COMMENTED** [review]
Minor change then good to go

File: soc/microchip/mec/Kconfig (hunk 7 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/Kconfig]: done

File: soc/microchip/mec/Kconfig.defconfig (hunk 8 above)
Comment (fabiobaltieri) [reply soc/microchip/mec/Kconfig.defconfig:6]: done

**Review by nordicjm — APPROVED** [review]

File: soc/microchip/mec/common/CMakeLists.txt [hunk 9]
```diff
@@ -12,7 +12,7 @@ endif()
 if (DEFINED CONFIG_MCHP_MEC_UNSIGNED_HEADER)
   set(MCHP_MEC_BIN_NAME ${CONFIG_KERNEL_BIN_NAME}.mchp.bin)
   set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
-    COMMAND ${PYTHON_EXECUTABLE} ${SOC_DIR}/${ARCH}/${SOC_FAMILY}/common/spigen/mec_spi_gen.py
+    COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/soc/microchip/mec/common/spigen/mec_spi_gen.py
```
Comment (tejlmand) [inline soc/microchip/mec/common/CMakeLists.txt:15]: No change requested, but want to point out that the following code resolves to same location:
```suggestion
    COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_LIST_DIR}/spigen/mec_spi_gen.py
```

**Review by tejlmand — APPROVED** [review]

**Pull request merged**